### DISCUSSION TITLE
Migrate MobiFlight to .NET 5 or newer

### DISCUSSION THREAD

**Discussion by neilenns** [MobiFlight/MobiFlight-Connector#508] [General]
Discussion related to #484

**Comment by MobiFlight-Admin** [comment]
Yeah!

**Reply by gyssels** [reply]
@danecreekphotography @MobiFlight-Admin 
**Please can you create the branch**: **preNet5** (you can name this one as you see fit) with code of main and, where I can PR to for your review.

**Reply by gyssels** [reply]
We will make it upgradable to .Net 4.8 first as you have libraries that are not .net 5 compatible. Migration will be easier with the latest .net framework. This will allow us to clean the clutter and undust.

**Reply by neilenns** [reply]
I don't believe I have permissions to create branches in this repo, but... why do you need one? Just do the work in your fork in a branch then open the PR against the main branch of the repo. Then it can be reviewed and merged as appropriate.

**Reply by gyssels** [reply]
I do in my fork, but a central place where all can push for the migration is not a bad idea, isn't it?

**Reply by neilenns** [reply]
Ah, I see what you mean, if there are multiple people working on the migration at the same time. To be honest I think you'll be the only one tackling it. Either way, I don't have permission so this is a question for Seb :)

**Reply by gyssels** [reply]
I will do start on my side then and when you guys are ready we PR back to yours then.

**Reply by MobiFlight-Admin** [reply]
Hey guys! This week i am extremely busy at work with no real option to support. 

I would suggest to first discuss a general strategy for migrating.

Can somebody please list the dependencies for all the libs we're using and check if they offer a .NET Core compatible NUGET version? FSUIPC Lib is not really Open Source if i am correct.

I also know that some UI elements are not supported by .NET Core (e.g. DataGridView, which I hate anyways and would like to get rid of).

Taking the time to create such an overview and forge a plan afterwards makes sense before we start changing things

**Reply by neilenns** [reply]
I'm just an innocent bystander on this one 😂 With the firmware PlatformIO migration, input shift registers, and my hacking on STM32 I'm just observing this one till it's done and I can do the build automation.

**Reply by gyssels** [reply]
@MobiFlight-Admin **Just ignore the creation of the branch**, I created a folder for .net5 so when I'll PR, I will be invisible to others users... we can reside together.

I am busy with CommandManager now, there is a nasty dependency with the forms on mobiflightconnector

by the way, you guys are crazy to accept PR on main! you should have a **"develop"** branch...

**Reply by neilenns** [reply]
If we're gonna debate branching strategies it better be over beer and snacks somewhere!

**Reply by gyssels** [reply]
**FEEDBACK CommandMessenger** I just have a deep look for CommandMessenger. the compilation doesn't go well

The project will only compile targeting the ".Net5.0-windows", that is a subset of the 
“.Net5.0”. Libraries like System.IO.Ports targets “.Net5.0”.
".Net5.0-windows" can call “.Net5.0”, and “.Net5.0” can’t call ".Net5.0-windows".
This issue make,s for example, the need to modify the tests projects as windows forms application to be able to run the tests, but I cannot guarantee its works, as I have exceptions in VisualStudio test runner that specs “.Net5.0” and not “.Net5.0-windows”

It can be an issue if MobiFlight cannot have a standard test suite…

Besides, the modification to CommandMessenger will include a substitution of the calls to libraries that are deprecated/obsolete in .Net5 and throws an exception:

BinaryFormatter.Serialize   (Microsoft docs states to change code to use JsonSerializer ou XmlSerializer, an extension of modifications rest to be determined)
BinaryFormatter.Deserialize
Thread.Abort() (not an equivalent, the thread approach should be revisited)

Conclusion:
As you probably know CmdMensager is a dead library (as that has its last updates in 2017, with a bunch of PR are abandoned between 2017-2021).
If (it's viable to migrate) {
    1)	Major overhaul of a “dead” library
    2)	Full depart from the CmdMenssager
    3)	Probably needs its own GitHub depot
}
Else If (it's NOT viable to migrate){
    1)	hunt for another **maintained** library (maybe one of your Arduinos guys can pinpoint one…)
}

**Reply by gyssels** [reply]
**FEEDBACK FSUIPCClient** NuGet for FSUIPC Client .Net5 available
